primer/src/Primer/Primitives.hs

```diff
@@ -276,6 +333,49 @@ primFunDef def args = case def of
     [exprToNat -> Just n] ->
       Right $ int $ fromIntegral n
     _ -> err
+  Animate -> case args of
+    -- Since we can only translate a `Picture` expression to an image once it is in normal form,
+    -- this guard will only pass when `picture` has no free variables other than `time`.
+    [PrimCon () (PrimInt duration), Lam () time picture]
+      | Just frames <- traverse diagramAtTime [0 .. duration * 100 `div` frameLength] ->
```

Comment:
(Pedantic off-by-one): I think this length of frame indices is one too long (not that a human would be able to visually tell unless the construction function was wild)!